project/TestParallelization.scala

```diff
@@ -50,17 +52,24 @@ object TestParallelization {
     Test / forkTestJVMCount := {
       sys.env.get("TEST_PARALLELISM_COUNT").map(_.toInt).getOrElse(4)
     },
+    Test / shardId := {
+      sys.env.get("SHARD_ID").map(_.toInt)
```

Comment:
can you use `numShards` here?

Reply:
No, that wouldn't be correct, this is supposed to be the shard ID assigned to a given action.
`numShards` is just used for the modulo math to determine if a test is assigned to the given shard ID or not. I should probably comment this, or if there's better names, open to suggestion.

Reply:
Oh, sorry, I just misread this. Ignore this comment. Thanks!